litestar/plugins/sqlalchemy.py

```diff
@@ -1,54 +1,31 @@
-from advanced_alchemy import filters, types
-from advanced_alchemy.base import (
-    AuditColumns,
-    BigIntAuditBase,
-    BigIntBase,
-    BigIntPrimaryKey,
-    CommonTableAttributes,
-    UUIDAuditBase,
-    UUIDBase,
-    UUIDPrimaryKey,
-    orm_registry,
-)
-from advanced_alchemy.extensions.litestar import (
-    AlembicAsyncConfig,
-    AlembicCommands,
-    AlembicSyncConfig,
-    AsyncSessionConfig,
-    EngineConfig,
-    SQLAlchemyAsyncConfig,
-    SQLAlchemyDTO,
-    SQLAlchemyDTOConfig,
-    SQLAlchemyInitPlugin,
-    SQLAlchemyPlugin,
-    SQLAlchemySerializationPlugin,
-    SQLAlchemySyncConfig,
-    SyncSessionConfig,
-)
+from litestar.utils import warn_deprecation
 
-__all__ = (
-    "orm_registry",
-    "filters",
-    "types",
-    "AuditColumns",
-    "BigIntAuditBase",
-    "BigIntBase",
-    "UUIDAuditBase",
-    "UUIDPrimaryKey",
-    "CommonTableAttributes",
-    "UUIDBase",
-    "BigIntPrimaryKey",
-    "AlembicCommands",
-    "AlembicAsyncConfig",
-    "AlembicSyncConfig",
-    "AsyncSessionConfig",
-    "SyncSessionConfig",
-    "SQLAlchemyDTO",
-    "SQLAlchemyDTOConfig",
-    "SQLAlchemyAsyncConfig",
-    "SQLAlchemyInitPlugin",
-    "SQLAlchemyPlugin",
-    "SQLAlchemySerializationPlugin",
-    "SQLAlchemySyncConfig",
-    "EngineConfig",
-)
+
+def __getattr__(attr_name: str) -> object:
+    from advanced_alchemy.extensions import litestar as aa  # pyright: ignore[reportMissingImports]
+
+    if attr_name in {
+        "AuditColumns",
+        "BigIntAuditBase",
+        "BigIntBase",
+        "BigIntPrimaryKey",
+        "CommonTableAttributes",
+        "UUIDAuditBase",
+        "UUIDBase",
+        "UUIDPrimaryKey",
+        "orm_registry",
+    }:
+        warn_deprecation(
+            deprecated_name=f"litestar.plugins.sqlalchemy.{attr_name}",
+            version="2.8.4",
```

Comment:
We've been holding back new deprecations for minor releases - there was at least one PR that we kept out of 2.8.3 that was only a refactor, but included a new depreciation, so we held it back for 2.9 - this would fall in the same category.

Reply:
Thanks.  Updated.